### PR TITLE
Fixed undefined id issue for simple product

### LINKED ISSUE #2757: Inconsistent behaviour of picture slider od PDP
## Current behavior
Picture slider mechanism needs to be triggered by 1st use of side arrows icons. Until then, clicking on color buttons or small icons in the bottom of the picture, will result in simple picture swtich, without sliding animation. After first use, every method of picture browsing will be using sliding animation.  

## Expected behavior
When using arrow buttons - animation should always be played
When using color icons - pictures should always be switched without animation

## Steps to reproduce the issue
1. Go to https://next.storefrontcloud.io/
2. Open procudt page with few pictures available
3. Switch between pictures using color icons
4. Switch between pictures using arrow icons on the sides
5. Now again switch between pictures using color icons (or small circle icons in the bottom)

See the video:
Slider Trigger

## Can you handle fixing this bug by yourself?

- [ ] YES
- [x] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [x] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details
- Browser: chrome, Opera, FF, Edge
- OS: Win10

### PR DESCRIPTION
**Pull request by vishal-7037**
### Related issues

closes #2757 

### Short description and why it's useful

- Fixed undefined id issue for simple product

### Screenshots of visual changes before/after (if there are any)

- When I opened any simple product following error occurred in developer console.
2019_07_09_Selection_001

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

**Comment by vishal-7037** [comment]
Hello,

Changes are added for Inconsistent behaviour of picture slider of product detail page. Please refer PR https://github.com/DivanteLtd/vue-storefront/pull/3015

Thank you.

File: src/themes/default/components/core/ProductGalleryCarousel.vue [hunk 1]
```diff
@@ -91,8 +91,10 @@ export default {
   mounted () {
     this.selectVariant()
 
-    const {color} = this.configuration
-    this.currentColor = color.id
+    if (this.configuration.hasOwnProperty('color')) {
```
Comment (patzick) [inline src/themes/default/components/core/ProductGalleryCarousel.vue]:
```suggestion
    if (this.configuration.color) {
```

**Comment by patzick** [comment]
@vishal-7037 when you create PR please always check this
image

please apply this suggestion and we can merge:)

**Comment by vishal-7037** [comment]
> @vishal-7037 when you create PR please always check this
> image
> 
> please apply this suggestion and we can merge:)

@patzick , Sure . I'll take care of this.

**Comment by vishal-7037** [comment]
@patzick,

When i tried to merge develop branch to this branch got error during compilation.
> Becase in `Claims` module `ClaimsState` file is removed but not removed it from `action.ts` file
https://github.com/DivanteLtd/vue-storefront/blob/dea3bf828db8f650192bb32f3f7ec72c7ae2598b/src/modules/claims/store/actions.ts#L3
Can you please look in to that?

Thanks

File: src/themes/default/components/core/ProductGalleryCarousel.vue (hunk 1 above)
Comment (vishal-7037) [reply src/themes/default/components/core/ProductGalleryCarousel.vue]: @patzick ,

I've updated your suggested changes

**Comment by vishal-7037** [comment]
@patzick ,

I've updated your suggested changes.
Please review it and let me know.

Thank you.

**Comment by pkarw** [comment]
@przspa please take a look at this one (CR)

**Comment by vishal-7037** [comment]
> code is greate but we need merge it to release branch, @vishal-7037 can you fix it?

@przspa You mean `master` branch or `develop` branch?
Can you please tell me which branch i need to merge?

Thanks

**Comment by przspa** [comment]
> > code is greate but we need merge it to release branch, @vishal-7037 can you fix it?
> 
> @przspa You mean `master` branch or `develop` branch?
> Can you please tell me which branch i need to merge?
> 
> Thanks

@vishal-7037 please crate new branch from release and PR to release :) and add changes to 1.10.0-rc.2 in changelog,Thanks! :) 

> This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from release branch and create Pull Request 3. Stabilisation fix back to release.

**Comment by vishal-7037** [comment]
> > > code is greate but we need merge it to release branch, @vishal-7037 can you fix it?
> > 
> > 
> > @przspa You mean `master` branch or `develop` branch?
> > Can you please tell me which branch i need to merge?
> > Thanks
> 
> @vishal-7037 please crate new branch from release and PR to release :) and add changes to 1.10.0-rc.2 in changelog,Thanks! :)
> 
> > This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from release branch and create Pull Request 3. Stabilisation fix back to release.

@przspa ,
I've created a PR from release branch. Please review and let me know.

Thanks

**Comment by patzick** [comment]
Closing in favour of #3239

**Pull request closed without merging**